sirepo/simulation_db.py

```diff
@@ -771,31 +771,34 @@ def static_libs():
 
 
 @contextlib.contextmanager
-def tmp_dir(chdir=False, qcall=None):
+def tmp_dir(chdir=False, qcall=None, dirname=None):
```

Comment:
I think this should be a separate call. It doesn't need to be a contextlib because it isn't going to go away. It could be a lot simpler and clearer to use that way.

Reply:
OK - I've added it as a new api: simulation_db.cache_dir(). Please take a look.